vector/src/main/java/im/vector/riotx/core/di/VectorComponent.kt

```diff
@@ -106,6 +107,8 @@ interface VectorComponent {
 
     fun vectorFileLogger(): VectorFileLogger
 
+    fun uiStateRepository(): UiStateRepository
```

Comment:
No need to be there yet, as you don't use singleton scope

Reply:
If I remove this, I get a compilation error :
```
di/ScreenComponent.java:57: error: [Dagger/MissingBinding] im.vector.riotx.features.ui.UiStateRepository cannot be provided without an @Provides-annotated method.
public abstract interface ScreenComponent {
                ^
      im.vector.riotx.features.ui.UiStateRepository is provided at
          im.vector.riotx.core.di.ScreenComponent.uiStateRepository()
  It is also requested at:
      im.vector.riotx.features.home.HomeDetailViewModel_AssistedFactory(…, uiStateRepository, …)
  The following other entry points also depend on it:
      im.vector.riotx.core.di.ScreenComponent.inject(im.vector.riotx.features.home.HomeDetailFragment)


```

Reply:
ok, as discussed, we will rework the Dagger module in another small PR